examples/stable_diffusion/convert_sd_adapter_to_peft.py

```diff
@@ -282,9 +282,15 @@ def detect_adapter_type(keys: List[str]) -> PeftType:
             elif kohya_type == "hada_t1":
                 adapter_info[model_type][peft_key].hada_t1 = tensor
                 adapter_info[model_type][peft_key].rank = tensor.shape[0]
-            elif kohya_type == "hada_t2":
+
+                # Right now there exists a bug, so we need to rewrite hada_t2 tensor with hada_t1 info
+                # https://github.com/KohakuBlueleaf/LyCORIS/pull/115
                 adapter_info[model_type][peft_key].hada_t2 = tensor
-                adapter_info[model_type][peft_key].rank = tensor.shape[0]
+
+            elif kohya_type == "hada_t2":
+                # adapter_info[model_type][peft_key].hada_t2 = tensor
```

Comment:
IIUC, once the bug is fixed, this code should be reverted to its previous state, is that right? How would we know if a user converts weights with vs without the bug?

Reply:
> IIUC, once the bug is fixed, this code should be reverted to its previous state, is that right?

Yes, but it will depend on whether the provided checkpoint was trained with or without this fix. There is no obvious way to determine it right now...

We may add a flag to this conversion script, but it will be on the user's side to decide what should be done here.

Reply:
Yeah, I think we need to have an option to switch between before vs after the patch. If the patch is tied to a specific version, we could add an argument where users pass the version and we check if it is from before vs after the patch.

Reply:
I added a CLI parameter to the conversion script to optionally enable/disable the fix through the weights.

> If the patch is tied to a specific version, we could add an argument where users pass the version and we check if it is from before vs after the patch.

As far as I see, current checkpoints out there do not contain information about LyCORIS version, so we probably have to manually force a user to provide, whether this fix should be applied or not (I also added a warning to highlight this problem).